src/meta/service/src/notification_service.rs

```diff
@@ -197,8 +197,7 @@ impl NotificationServiceImpl {
         match &self.metadata_manager {
             MetadataManager::V1(mgr) => {
                 let fragment_guard = mgr.fragment_manager.get_fragment_read_guard().await;
-                let worker_slot_mappings =
-                    fragment_guard.all_running_fragment_mappings().collect_vec();
+                let worker_slot_mappings = fragment_guard.all_running_fragment_mappings()?;
```

Comment:
I would prefer to add a warning and return empty mapping to FE, so that the catalogs can still be visible, dropped, cancelled etc and also be able to queried in singleton.